AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/AWSDataStorePlugin.swift

```diff
@@ -126,6 +128,8 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
                 try resolveStorageEngine(dataStoreConfiguration: dataStoreConfiguration)
                 try storageEngine.setUp(modelSchemas: ModelRegistry.modelSchemas)
                 try storageEngine.applyModelMigrations(modelSchemas: ModelRegistry.modelSchemas)
+                self.isStorageEngineInitialized.set(true)
```

Comment:
`DataStore.stop` and `DataStore.clear` will deallocate the storageEngine so I'm thinking this variable should be set back to `false` somewhere in those code paths